OpenEdXMobile/src/main/java/org/edx/mobile/view/LockedCourseUnitFragment.kt

```diff
@@ -0,0 +1,77 @@
+package org.edx.mobile.view
+
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.inject.Inject
+import de.greenrobot.event.EventBus
+import org.edx.mobile.R
+import org.edx.mobile.databinding.FragmentLockedCourseUnitBinding
+import org.edx.mobile.event.CourseUpgradeDataReceivedEvent
+import org.edx.mobile.model.api.CourseUpgradeResponse
+import org.edx.mobile.model.api.EnrolledCoursesResponse
+import org.edx.mobile.model.course.CourseComponent
+import org.edx.mobile.module.analytics.Analytics
+import org.edx.mobile.module.analytics.AnalyticsRegistry
+
+class LockedCourseUnitFragment : CourseUnitFragment() {
+    @Inject
+    var analyticsRegistry: AnalyticsRegistry? = null
+
+    companion object {
+        @JvmStatic
+        fun newInstance(unit: CourseComponent,
+                        courseData: EnrolledCoursesResponse,
+                        courseUpgradeData: CourseUpgradeResponse?): LockedCourseUnitFragment {
+            val fragment = LockedCourseUnitFragment()
+            val bundle = Bundle()
+            bundle.putSerializable(Router.EXTRA_COURSE_UNIT, unit)
+            bundle.putSerializable(Router.EXTRA_COURSE_DATA, courseData)
+            courseUpgradeData?.let {
+                bundle.putSerializable(Router.EXTRA_COURSE_UPGRADE_DATA, courseUpgradeData)
+            }
+            fragment.arguments = bundle
+            return fragment
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val binding: FragmentLockedCourseUnitBinding =
+                DataBindingUtil.inflate(inflater, R.layout.fragment_locked_course_unit, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val courseUpgradeData = arguments?.getSerializable(Router.EXTRA_COURSE_UPGRADE_DATA)
+        if (courseUpgradeData != null) {
+            val courseData = arguments?.getSerializable(Router.EXTRA_COURSE_DATA) as EnrolledCoursesResponse
+            loadPaymentBannerFragment(courseData, courseUpgradeData as CourseUpgradeResponse)
```

Comment:
code can be replaced by the followings line
```
        val courseUpgradeData = arguments?.getSerializable(Router.EXTRA_COURSE_UPGRADE_DATA) as CourseUpgradeResponse?
        if (courseUpgradeData != null) {
            val courseData = arguments?.getSerializable(Router.EXTRA_COURSE_DATA) as EnrolledCoursesResponse
            loadPaymentBannerFragment(courseData, courseUpgradeData)
        }
```

Reply:
done